rpc/handlers.go

```diff
@@ -1039,24 +1047,43 @@ func (h *Handler) Call(call FunctionCall, id BlockID) ([]*felt.Felt, *jsonrpc.Er
 	return res, nil
 }
 
-func (h *Handler) TransactionStatus(hash felt.Felt) (Status, *jsonrpc.Error) {
-	var status Status
+func (h *Handler) TransactionStatus(hash felt.Felt) (*TransactionStatus, *jsonrpc.Error) {
+	var status *TransactionStatus
```

Comment:
```suggestion
	var status TransactionStatus
```

and you dont need to `new`` it in `ErrTxnHashNotFound` case

Reply:
The status is now a struct. If I'm missing something, let me know.